packages/netlify-cms-core/src/actions/editorialWorkflow.ts

```diff
@@ -121,138 +119,109 @@ function unpublishedEntriesFailed(error: Error) {
   };
 }
 
-function unpublishedEntryPersisting(
-  collection: Collection,
-  entry: EntryMap,
-  transactionID: string,
-) {
+function unpublishedEntryPersisting(collection: Collection, slug: string) {
   return {
     type: UNPUBLISHED_ENTRY_PERSIST_REQUEST,
     payload: {
       collection: collection.get('name'),
-      entry,
+      slug,
     },
-    optimist: { type: BEGIN, id: transactionID },
   };
 }
 
-function unpublishedEntryPersisted(collection: Collection, transactionID: string, slug: string) {
+function unpublishedEntryPersisted(collection: Collection, entry: EntryMap) {
   return {
     type: UNPUBLISHED_ENTRY_PERSIST_SUCCESS,
     payload: {
       collection: collection.get('name'),
-      slug,
+      entry,
     },
-    optimist: { type: COMMIT, id: transactionID },
   };
 }
 
-function unpublishedEntryPersistedFail(error: Error, transactionID: string) {
+function unpublishedEntryPersistedFail(error: Error, collection: Collection, slug: string) {
   return {
     type: UNPUBLISHED_ENTRY_PERSIST_FAILURE,
-    payload: { error },
-    optimist: { type: REVERT, id: transactionID },
+    payload: {
+      error,
+      collection,
```

Comment:
```suggestion
      collection: collection.get('name'),
```

We should pass the collection name here https://github.com/netlify/netlify-cms/pull/4289/files#diff-afa8a4f2b5ad4d666122939a5978feb7bf4eacbc4810c48bb6903189bad3c9a7R98

Reply:
> We should pass the collection name here https://github.com/netlify/netlify-cms/pull/4289/files#diff-afa8a4f2b5ad4d666122939a5978feb7bf4eacbc4810c48bb6903189bad3c9a7R98

I did the change for action part (`collection: collection.get('name'),`)
But I didn't understand what should I do for the reducer part and didn't find any example like `collection.name`.
Could you give an example of that part?

Reply:
Sorry, I just referenced the reducer to make it clear it uses a string. You don't need to change anything there